### PR TITLE
Refactor vipNetworkList to include CIDR

### PR DESCRIPTION
**Pull request by chauhanshubham**
This updates the VipNetworkList schemas in helm's values.yaml,
AviInfraSetting CRD and ako-operator's AKOConfig, which now includes
an extra `cidr` field that allows users to choose specific subnet in
a network. This replaces the previous subnetIP/subnetPrefix fields.
The AviInfraSetting and values.yaml route has validations for the CIDR
string to be of a certain format.
This also removes unused EastWest fields across AKO.

### CODE REVIEW

**Comment by chauhanshubham** [comment]
@sudswasavi @monotosh-avi could you please check the updated schema and see if it looks good?

File: internal/nodes/avi_model_nodes.go [hunk 1]
```diff
@@ -909,12 +909,9 @@ type AviVSVIPNode struct {
 	Tenant                  string
 	CloudConfigCksum        uint32
 	FQDNs                   []string
-	EastWest                bool
 	VrfContext              string
 	IPAddress               string
-	SubnetIP                string
-	SubnetPrefix            int32
-	NetworkNames            []string
+	VipNetworks             []akov1alpha1.AviInfraSettingVipNetwork
```
Comment (monotosh-avi) [inline internal/nodes/avi_model_nodes.go:914]: Directly referring a CRD field here may not be required. We can create a struct with two strings and use it. 
Just a suggestion, functionality wise it is fine

**Review by monotosh-avi — COMMENTED** [review]
Looks good to me. 
@aaha97  can you also take a look it this spl from the point view of AWS ?

**Review by aaha97 — DISMISSED** [review]
lgtm

**Review by monotosh-avi — APPROVED** [review]

File: internal/lib/constants.go [hunk 2]
```diff
@@ -114,6 +114,7 @@ const (
 	SyncStatusKey                              = "syncstatus"
 	NoFreeIPError                              = "No available free IPs"
 	ConfigDisallowedDuringUpgradeError         = "Configuration is disallowed during upgrade"
+	IPCIDRRegex                                = `^(\b([01]?[0-9][0-9]?|2[0-4][0-9]|25[0-5]).){3}[0-9]{1,3}\/(([0-9]|[1-2][0-9]|3[0-2]))?$`
```
[the remark this answers is not in the document]
Comment (chauhanshubham) [reply internal/lib/constants.go:117]: yes thanks! was also missing a backslash for `\.`

**Review by monotosh-avi — APPROVED** [review]

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**